flusher/flusher.go

```diff
@@ -0,0 +1,61 @@
+package flusher
+
+import(
+	"time"
+
+	"github.com/drone/drone-cache-lib/storage"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type dirtyFunc func(storage.FileEntry) bool
+
+type Flusher struct {
+	store storage.Storage
+	dirty func(storage.FileEntry) bool
+}
+
+func New(s storage.Storage, fn dirtyFunc) Flusher {
+	return Flusher{ store: s, dirty: fn }
+}
+
+func NewDefault(s storage.Storage) Flusher {
+	return Flusher{ store: s, dirty: TimeOp() }
+}
+
+func (f *Flusher) Flush(src string) error {
+	log.Infof("Cleaning files from %s", src)
+
+	files, err := f.store.List(src)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		if f.dirty(file) {
+			err := f.store.Delete(file.Path)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func TimeOp() dirtyFunc {
+	return func(file storage.FileEntry) bool {
+		// Do not delete directories
```

Comment:
How do I then assign the function in `NewDefaultFlusher` @ https://github.com/drone/drone-cache-lib/blob/7966928e65d26567af5eded3326bd01316b031dc/cache/flusher.go#L23 ?